### PR TITLE
Initial support for LLMNR

### PR DESCRIPTION
**Pull request by defensivedepth**
## Proposed changes

Adds support for Canary LLMNR. Uses Scapy to broadcast a LLMNR query for canary hostname. If it receives a LLMNR response, it confirms that it was for the canary hostname and then logs the event. The following is configurable:
- LLMNR Port (TCP/5355 by default)
- Canary LLMNR Hostame
- How often to broadcast the LLMNR query (in seconds)
- Splay time to add randomness to the broadcast (in seconds)

Sample log output, generated by using Responder to poison LLMNR request:

`{"dst_host": "0.0.0.0", "dst_port": 5355, "local_time": "2024-01-20 21:41:58.716469", "local_time_adjusted": "2024-01-20 21:41:58.716499", "logdata": {"query_hostname": "DC03", "response": "DNS Ans \"10.0.0.22\" "}, "logtype": 19001, "node_id": "opencanary-1", "src_host": "192.168.16.27", "src_port": 5355, "utc_time": "2024-01-20 21:41:58.716495"}`

Discussion: https://github.com/thinkst/opencanary/discussions/335

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update

## Checklist

I will add docs once it's confirmed that this PR will be accepted. 

- [ ] Lint and unit tests pass locally with my changes (if applicable)
- [ ] I have run pre-commit (`pre-commit` in the repo)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] Linked to the relevant github issue or github discussion

## Further comments

N/A

### CODE REVIEW

**Comment by defensivedepth** [comment]
Anything I can do to help move this along?

**Comment by jayjb** [comment]
Hi @defensivedepth, @theidiotyouyellat,

Sorry for the delay with this. We like the idea, we don't love the dependency on Scapy though (its quite a heavy dependency for what we using it for). Im trying to figure a way we can craft those Queries without it. If not, I'll likely accept this for now while we improve it.

**Comment by defensivedepth** [comment]
@jayjb Thanks for the feedback. The Scapy dep is not new - the SNMP module already requires it. What specifically is the concern?

**Comment by mclmax** [comment]
hey @defensivedepth and @theidiotyouyellat, taking a look. hit you back shortly

File: bin/opencanary.tac [hunk 1]
```diff
@@ -22,7 +22,7 @@ from opencanary.modules.git import CanaryGit
 from opencanary.modules.redis import CanaryRedis
 from opencanary.modules.tcpbanner import CanaryTCPBanner
 from opencanary.modules.rdp import CanaryRDP
-
+from opencanary.modules.llmnr import CanaryLLMNR
```
Comment (mclmax) [inline bin/opencanary.tac:25]: Could we copy how we handle the imports for the the SNMP module here? [SNMP import](https://github.com/thinkst/opencanary/blob/5b6dbd01b9cc404f8d5b96f8e9be8b620189078a/bin/opencanary.tac#L59)

Reading the docs it looks like we have a weak dependency on Scapy. It doesn't have to be installed. So this change will just make sure folks pulling a newer version won't have the thing falling over if they didn't have Scapy installed originally (and might not use this new service).

**Review by mclmax — CHANGES_REQUESTED** [review]
Thanks @defensivedepth, the code looks great! Sorry about the delay here getting our feedback. We'd love to have this in. I've just left a request for a small change to how we handle the import and we should be good to go!

**Comment by defensivedepth** [comment]
I resolved the change request, thanks!

**Review by mclmax — APPROVED** [review]

**Comment by mclmax** [comment]
awesome, thanks @defensivedepth! merged

**Comment by mclmax** [comment]
Now it's merged ;)

**Pull request merged**